### DISCUSSION TITLE
Ghostscript pdf not recognised

### DISCUSSION THREAD

**Discussion by grego1981** [pymupdf/PyMuPDF#1948] [Q&A] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
I have a flock of PDFs that are have in the following attributes:
Producer: GPL Ghostscript 9.15
PDF Version: 1.4
That are not recognised using PDF2Text.py form utilities and returns garbage data.
## To Reproduce (mandatory)
Run PDF2Text.py the produced text has nothing to do with the pdf data.

## Expected behavior (optional)
Get the pdf text

## Your configuration (mandatory)
 - Ubuntu 20.04
 - Python version, Python 3.8.10
 - PyMuPDF version pymupdf-1.20.2, installation method (**wheel**  ).

print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)
3.8.10 (default, Jun 22 2022, 20:18:18)
[GCC 9.4.0]
linux

PyMuPDF 1.20.2: Python bindings for the MuPDF 1.20.3 library.
Version date: 2022-08-13 00:00:01.
Built for Python 3.8 on linux (64-bit).

Attached the example pdf
[example.pdf](https://github.com/pymupdf/PyMuPDF/files/9696910/example.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
I am sorry, but this is a MuPDF problem ... if it is a problem at all! The file uses a so-called Type3 font and contains no information for back-translating glyphs to unicode values.
The `/Encoding` object xref 107 would have to contain this info - which it doesn't.
The only thing you can do is OCR-ing ...

**Comment by grego1981** [comment]
Thank you for the prompt reply! So in this format is it possible to use the replace font scripts? I've tried OCRing it but there are many random characters where lines are close to the letters... although the text output is very close by 90%..

**Comment by JorjMcKie** [comment]
> Thank you for the prompt reply! So in this format is it possible to use the replace font scripts? I've tried OCRing it but there are many random characters where lines are close to the letters... although the text output is very close by 90%..

Unfortunately not, because font replacement also internally uses text extraction 😒.
You may influence the OCR quality by using a higher resolution, works in ocrmypdf as well as with PyMuPDF. Both are based on Tesseract anyway.

**Comment by grego1981** [comment]
> > Thank you for the prompt reply! So in this format is it possible to use the replace font scripts? I've tried OCRing it but there are many random characters where lines are close to the letters... although the text output is very close by 90%..
> 
> Unfortunately not, because font replacement also internally uses text extraction 😒. You may influence the OCR quality by using a higher resolution, works in ocrmypdf as well as with PyMuPDF. Both are based on Tesseract anyway.

I'll try that as soon as possible!

**Comment by JorjMcKie** [comment]
I am going to convert this to a Discussions item, so you may add any further observation to it later on.
